app/services/provider_interface/save_provider_user_service.rb

```diff
@@ -16,6 +18,13 @@ def call!
 
   private
 
+    def assert_permissions_for_providers!
+      authorisation = ProviderAuthorisation.new(actor: @actor)
+      return if @wizard.provider_permissions.keys.all? { |provider_id| authorisation.can_manage_users_for?(Provider.find(provider_id)) }
+
+      raise ProviderAuthorisation::NotAuthorisedError, 'You are not allowed to add users to these providers'
+    end
+
     def email_exists?
       ProviderUser.find_by(email_address: wizard.email_address).present?
```

Comment:
This will fail if a mixed-case email address has been entered.

Reply:
Good point. Will downcase like we do in other 'forms'.